odoo/addons/base/models/ir_qweb.py

```diff
@@ -295,45 +298,71 @@ def get_asset_bundle(self, xmlid, files, env=None):
         'xml' not in tools.config['dev_mode'],
         tools.ormcache_context('xmlid', 'options.get("lang", "en_US")', 'css', 'js', 'debug', 'async_load', 'defer_load', 'lazy_load', keys=("website_id",)),
     )
-    def _get_asset_nodes(self, xmlid, options, css=True, js=True, debug=False, async_load=False, defer_load=False, lazy_load=False, values=None):
-        files, remains = self._get_asset_content(xmlid, options, css, js)
+    def _get_asset_nodes(self, xmlid, options, css=True, js=True, debug=False, async_load=False, defer_load=False, lazy_load=False, media=None, values=None):
+        nodeAttrs = None
+        if css and media:
+            nodeAttrs = {
+                'media': media,
+            }
+        files, remains = self._get_asset_content(xmlid, options, nodeAttrs)
         asset = self.get_asset_bundle(xmlid, files, env=self.env)
+        remains = [node for node in remains if (css and node[0] == 'link') or (js and node[0] == 'script')]
```

Comment:
Is this relevant? All paths and URLs should already be filtered in `get_addon_files`

Reply:
@Arcasias 
I believe it is relevant, but not linked to your algo by itself.

TLDR: for performance reasons, we ormcahe a max of stuff, we need to go your algo only once, with both js and css to true; the heuristic on the file extension is then necessary

What was done before, and that we tried to reproduce (this is a performance decision that needs to be disputed :D ) is that we go through the asset only once (we go through your algo only once) with css and js to True.

The main parent function call to your algo is _get_asset_nodes ; which is orrmcached on the booleans css and js. 

Since _get_asset_content is not ormcached on js and css booleans, it needs to have done the filtering work for both css and js at the same time, hence the heuristic on the file extension.

_get_asset_nodes takes the result of _get_asset_content (which is now a constant containing both js and css) and just has to apply the "remain" filter on it (twice : one for js and one for css)

Reply:
Right... did not see the `js` and `css` arguments both to true :D
Alright then, fair point.